app/components/allContacts.js

```diff
@@ -0,0 +1,34 @@
+import { useState } from 'react';
+import { ContactsTable } from './contactsTable'
+import { useContacts } from '../contexts/ContactsContext';
+
+export function AllContacts() {
+  const { contacts = [] } = useContacts();
+  const [searchTerm, setSearchTerm] = useState('');
+  const [searchResult, setSearchResult] = useState([]);
+  
+  const handleSearchChange = (event) => {
+    setSearchTerm(event.target.value)
+  };
+
+  const handleSearchResult = (event) => {
+    const nameResult = contacts.filter((contact) => 
+      contact.name.toLowerCase() === event.target.value.toLowerCase()

```

Comment:
```suggestion
      contact?.name?.toLowerCase() === event.target.value.toLowerCase()

```
This can help you prevent crashing the app in case you don't have a name.